cmd/catchpointdump/bench.go

```diff
@@ -0,0 +1,184 @@
+		}
+		loadOnly = true
+		benchmark := makeBenchmarkReport()
+
+		if catchpointFile == "" {
+			if round == 0 {
+				return fmt.Errorf("round not set")
+			}
+			stage := benchmark.startStage("network")
+			catchpointFile, err = downloadCatchpointFromAnyRelay(networkName, round, relayAddress)
+			if err != nil {
+				return fmt.Errorf("failed to download catchpoint : %v", err)
+			}
+			stage.completeStage()
+		}
+		stats, err := os.Stat(catchpointFile)
+		if err != nil {
+			return fmt.Errorf("unable to stat '%s' : %v", catchpointFile, err)
+		}
+
+		catchpointSize := stats.Size()
+		if catchpointSize == 0 {
+			return fmt.Errorf("empty file '%s' : %v", catchpointFile, err)
+		}
+
+		genesisInitState := ledgercore.InitState{
+			Block: bookkeeping.Block{BlockHeader: bookkeeping.BlockHeader{
+				UpgradeState: bookkeeping.UpgradeState{
+					CurrentProtocol: protocol.ConsensusCurrentVersion,
+				},
+			}},
+		}
+		cfg := config.GetDefaultLocal()
+		l, err := ledger.OpenLedger(logging.Base(), "./ledger", false, genesisInitState, cfg)
+		if err != nil {
+			return fmt.Errorf("unable to open ledger : %v", err)
+		}
+
+		defer os.Remove("./ledger.block.sqlite")
+		defer os.Remove("./ledger.block.sqlite-shm")
+		defer os.Remove("./ledger.block.sqlite-wal")
+		defer os.Remove("./ledger.tracker.sqlite")
+		defer os.Remove("./ledger.tracker.sqlite-shm")
+		defer os.Remove("./ledger.tracker.sqlite-wal")
+		defer l.Close()
+
+		catchupAccessor := ledger.MakeCatchpointCatchupAccessor(l, logging.Base())
+		err = catchupAccessor.ResetStagingBalances(context.Background(), true)
+		if err != nil {
+			return fmt.Errorf("unable to initialize catchup database : %v", err)
+		}
+
+		reader, err := os.Open(catchpointFile)
+		if err != nil {
+			return fmt.Errorf("unable to read '%s' : %v", catchpointFile, err)
+		}
+		defer reader.Close()
+
+		printDigests = false
+		stage := benchmark.startStage("database")
+
+		_, err = loadCatchpointIntoDatabase(context.Background(), catchupAccessor, reader, catchpointSize)
+		if err != nil {
+			return fmt.Errorf("unable to load catchpoint file into in-memory database : %v", err)
+		}
+		stage.completeStage()
+
+		stage = benchmark.startStage("digest")
+
+		err = buildMerkleTrie(context.Background(), catchupAccessor)
+		if err != nil {
+			return fmt.Errorf("unable to build Merkle tree : %v", err)
+		}
+		stage.completeStage()
+
+		benchmark.printReport()
+		if reportJsonPath != "" {
+			if err := benchmark.saveReport(reportJsonPath); err != nil {
+				fmt.Printf("error writing report to %s: %v\n", reportJsonPath, err)
+			}
+		}
+
+		return err
+	},
+}
+
+func downloadCatchpointFromAnyRelay(network string, round int, relayAddress string) (string, error) {
+	var addrs []string
+	if relayAddress != "" {
+		addrs = []string{relayAddress}
+	} else {
+		//append relays
+		dnsaddrs, err := tools.ReadFromSRV(context.Background(), "algobootstrap", "tcp", networkName, "", false)
```

Comment:
"algobootstrap" probably should not be here since they not obliged to have catchpoints except few most recent ones.